app/src/main/java/io/neurolab/activities/MeditationActivity.java

```diff
@@ -75,13 +76,28 @@ private void grabNecessaryReferencesAndSetListeners() {
         progressTimeView = findViewById(R.id.progress_time);
         durationView = findViewById(R.id.duration_view);
         trackNameView = findViewById(R.id.track_name);
+        pauseButton.setVisibility(View.GONE); // The pause button should be invisible initially
 
+        //Handling Play-Pause-Reset buttons according to the player state.
         pauseButton.setOnClickListener(
-                view -> playerAdapter.pause());
+                view -> {
+                    playerAdapter.pause();
+                    pauseButton.setVisibility(View.GONE);
+                    playButton.setVisibility(View.VISIBLE);
+                });
+
         playButton.setOnClickListener(
-                view -> playerAdapter.play());
+                view -> {
+                    playerAdapter.play();
+                    pauseButton.setVisibility(View.VISIBLE);
+                    playButton.setVisibility(View.GONE);
+                });
         resetButton.setOnClickListener(
-                view -> playerAdapter.reset());
+                view -> {
+                    playerAdapter.reset();
+                    playButton.setVisibility(View.VISIBLE);
+                    pauseButton.setVisibility(View.INVISIBLE);
```

Comment:
@Sarthak2601 Why not `GONE`?

Reply:
Yes, we can do that as well. It seems that any of these will work so might as well use `INVISIBLE` only.

Reply:
@Sarthak2601 Please follow one pattern. VISIBLE/GONE OR VISIBLE/INVISIBLE. Also remember, in case of invisible, the view is actually there and takes that place whereas in case of gone it completely disappears from the layout

Reply:
Done